File: haskell/iniciante/1040.hs
-- title: Média 3
-- number of the problem: 1040
-- date: 22/09/20
-- author: ltsdw

import Text.Printf

main :: IO ()
main = do
    l1 <- getLine

    let (n1 : n2 : n3 : n4 : _) = map (read :: String -> Double) $ words l1

    let media = ( (n1 * 2.0) + (n2 * 3.0) + (n3 * 4.0) + (n4 * 1.0) ) / 10.0

    if media >= 7
        then
            printf "Media: %.1f\nAluno aprovado.\n" media

    else if media < 5
        then
            printf "Media: %.1f\nAluno reprovado.\n" media

    else if media >= 5 && media <= 6.9
        then do
            printf "Media: %.1f\nAluno em exame.\n" media

            exame' <- getLine

            let exame = read exame' :: Double

            let mediae = (exame + media) / 2.0

            printf "Nota do exame: %.1f\nAluno aprovado.\nMedia final: %.1f\n" exame mediae
    else
        return ()

File: python/iniciante/1040.py
'''
title: Média 3
number of the problem: 1040
date: 03/03/2018
author: ltsdw
'''

n1, n2, n3, n4 = map(float, input('').split())
media = ((n1*2) + (n2*3) + (n3*4) + (n4*1)) / 10

if media >= 7:
    print(f'Media: {media:.1f}\nAluno aprovado.')
if media < 5:
    print(f'Media: {media:.1f}\nAluno reprovado.')
if media >= 5 and media <= 6.9:
    print(f'Media: {media:.1f}\nAluno em exame.')
    exame = float(input(''))
    mediae = (exame + media) / 2
    print(f'Nota do exame: {exame:.1f}\nAluno aprovado.\nMedia final: {mediae:.1f}')

